GitCommands/RevisionReader.cs

```diff
@@ -240,19 +240,31 @@ private IReadOnlyCollection<GitRevision> GetRevisionsFromArguments(GitArgumentBu
             using IProcess process = _module.GitCommandRunner.RunDetached(cancellationToken, arguments, redirectOutput: true, outputEncoding: GitModule.LosslessEncoding);
             cancellationToken.ThrowIfCancellationRequested();
 
+            // Initial buffer to give very quick feedback to user
+            List<GitRevision> revisions = new(100);
+            const int commitsBatchSize = 25_000;
             foreach (ReadOnlyMemory<byte> chunk in process.StandardOutput.BaseStream.SplitLogOutput())
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                if (TryParseRevision(chunk.Span, out GitRevision? revision))
+                if (TryParseRevision(chunk, out GitRevision? revision))
                 {
 #if TRACE_REVISIONREADER
                     revisionCount++;
 #endif
-                    subject.OnNext(revision);
+                    revisions.Add(revision);
+                    if (revisions.Count == revisions.Capacity)
```

Comment:
In addition, we should use a timer here in case git very slowly yields revisions (e.g. when filtering using "Diff contains").
Can be added in a follow-up.